.github/workflows/archive.yml

```
@@ -0,0 +1,76 @@
+#
+# This source file is part of the Stanford Biodesign Digital Health Group open-source organization
+#
+# SPDX-FileCopyrightText: 2022 Stanford University and the project authors (see CONTRIBUTORS.md)
+#
+# SPDX-License-Identifier: MIT
+#
+
+name: Build XCArchive
+
+on:
+  workflow_call:
+    inputs:
+      workspaceFile:
+        description: 'The path of the Xcode Workspace file (including the file extension).'
+        type: string
+        required: true
+      XCArchiveName:
+        description: 'The name of the to be created XCArchive. It must not include the default XCArchive extension ".xcarchive" as the file naming is automatically done be the workflow.'
+        type: string
+        required: true
+      scheme:
+        description: 'The project scheme that should be used for the archiving process.'
+        type: string
+        required: true
+      version:
+          description: 'The version number of the XCFramework embedded in the XCArchives.'
+          type: string
+          required: true
+      configuration:
+        description: 'The build configuration to use when archiving the scheme, either Debug or Release.'
+        type: string
+        required: false
+        default: 'Release'
+      runsonlabels:
+          description: 'JSON-based collection of labels indicating which type of github runner should be chosen.'
+          required: false
+          type: string
+          default: '["macos-13"]'
+
+jobs:
+  build-xcarchive:
+    name: Build XCArchive
+    runs-on: ${{ fromJson(inputs.runsonlabels) }}
+    strategy:
+      matrix:
+        sdk: [iphoneos, iphonesimulator]
+    steps:
+    - uses: actions/checkout@v3
+    - uses: maxim-lobanov/setup-xcode@v1
+      with:
+        xcode-version: latest-stable
+    - name: Check environment
+      run: |
+          xcodebuild -version
+          swift --version
+          echo Release version: ${{ inputs.version }}
+          echo SDK: ${{ matrix.sdk }}
+    - name: Archive for iOS 
+      run: |
+          xcodebuild archive \
+            -workspace ${{ inputs.workspaceFile }} \
+            -scheme ${{ inputs.scheme }} \
+            -configuration ${{ inputs.configuration }} \
+            -sdk ${{ matrix.sdk }} \
+            -archivePath './.build/${{ inputs.XCArchiveName }}-${{ matrix.sdk }}.xcarchive' \
+            SKIP_INSTALL=NO \
+            BUILD_LIBRARY_FOR_DISTRIBUTION=YES \
+            ONLY_ACTIVE_ARCH=NO \
+            CI=TRUE \
+            ORK_FRAMEWORK_VERSION_NUMBER=${{ inputs.version }}
```

We might want to generalize this to ensure that this can be used in different packages and therefore adapt the ResearchKit fork.
```suggestion
            VERSION_NUMBER=${{ inputs.version }}
```